Add Hosted regions to resource reference (#1725)
astronomer/docs: astro/resource-reference-azure.md

## Default cluster values

> | Resource                                                                                                                 | Description                                                                                                                                                                                                                                                                               | Quantity/ Default Size                                                                        | Configurable? |

Author: ```suggestion
| Resource                                                                                                                 | Description                                                                                                                                                                                                                                                                               | Quantity/ Default Size                                                                        | Configurable  |
```

| Code               | Name               | Astro - Bring Your Own Cloud | Astro - Hosted |
| ------------------ | ------------------ | ---------------------------- | -------------- |
| `australiaeast`    | Australia East     | ✔️                            |                |
| `canadacentral`    | Canada Central     | ✔️                            | ✔️              |
| `centralindia`     | Central India      | ✔️                            |                |
| `centralus`        | Central US         | ✔️                            |                |
| `eastasia`         | East Asia          | ✔️                            |                |
| `eastus`           | East US            | ✔️                            |                |
| `japaneast`        | Japan East         | ✔️                            |                |
| `koreacentral`     | Korea Central      | ✔️                            |                |
| `northeurope`      | North Europe       | ✔️                            | ✔️              |
| `southafricanorth` | South Africa North | ✔️                            |                |

> | `southcentralus`   | South Central US   | ✔️                            |                |

Author: ```suggestion
| `southcentralus`   | South Central US   | ✔️                            |                |
| `southeastasia`   | South East Asia   | ✔️                            |    ✔️ (Private Preview)            |
```

| `uksouth`          | UK South           | ✔️                            |                |
| `westeurope`       | West Europe        | ✔️                            | ✔️              |

> | `westus3`          | West US 3          | ✔️                            |                |

Author: ```suggestion
| `westus2`          | West US 2         | ✔️                            |   ✔️ (Private preview)             |
| `westus3`          | West US 3          | ✔️                            |                |
```

> Modifying the region of an existing Astro cluster isn't supported. If you're interested in a region that isn't on this list, contact [Astronomer support](https://cloud.astronomer.io/support).

Reviewer: ```suggestion
Modifying the region of an existing Astro cluster isn't supported. If you're interested in a region that isn't listed, contact [Astronomer support](https://cloud.astronomer.io/support).
```
